File: frontend/src/api/accounts.ts
import request from './request'
import type { ResponseModel } from '@/types'

/**
 * 账号信息
 */
export interface Account {
  id: number
  phone: string
  company_id: number
  company_code: string
  company_name: string
  user_id: string | null
  last_login: string | null
  status: number
  created_at: string
  updated_at: string
}

/**
 * 创建/更新账号参数
 */
export interface CreateAccountParams {
  phone: string
  password: string
  company_code: string
  company_name: string
  domain?: string
}

/**
 * 更新账号参数
 */
export interface UpdateAccountParams {
  password?: string
}

/**
 * 获取所有账号
 */
export const getAccounts = (): Promise<ResponseModel<Account[]>> => {
  return request.get('/accounts')
}

/**
 * 获取单个账号
 */
export const getAccount = (id: number): Promise<ResponseModel<Account>> => {
  return request.get(`/accounts/${id}`)
}

/**
 * 创建或更新账号（Upsert）
 */
export const createOrUpdateAccount = (data: CreateAccountParams): Promise<ResponseModel<Account>> => {
  return request.post('/accounts', data)
}

/**
 * 更新账号密码
 */
export const updateAccount = (id: number, data: UpdateAccountParams): Promise<ResponseModel<Account>> => {
  return request.put(`/accounts/${id}`, data)
}

/**
 * 删除账号
 */
export const deleteAccount = (id: number): Promise<ResponseModel> => {
  return request.delete(`/accounts/${id}`)
}

/**
 * 手动登录账号
 */
export const loginAccount = (id: number): Promise<ResponseModel> => {
  return request.post(`/accounts/${id}/login`)
}

/**
 * 检查账号状态
 */
export const checkAccountStatus = (id: number): Promise<ResponseModel<{
  valid: boolean
  status: number
  last_login: string | null
}>> => {
  return request.get(`/accounts/${id}/status`)
}

/**
 * 公司信息
 */
export interface CompanyInfo {
  userId: string
  userType: number
  companyCode: string
  company: string
  domain: string
}

/**
 * 验证账号并获取公司列表
 */
export const checkAndGetUsers = (account: string, password: string): Promise<ResponseModel<{
  json: {
    code: number
    msg: string
    data: CompanyInfo[]
  }
  cookies: Record<string, string>
}>> => {
  return request.post('/yunke/check-users', { account, password })
}

File: backend/app/api/v1/accounts.py
"""账号管理API路由"""

from fastapi import APIRouter
from loguru import logger

from app.models.yunke_account import (
    YunkeAccountCreate,
    YunkeAccountResponse,
    YunkeAccountUpdate,
)
from app.schemas.response import ResponseModel
from app.services import account_service

router = APIRouter(prefix="/accounts", tags=["账号管理"])


@router.get("", response_model=ResponseModel[list[YunkeAccountResponse]])
async def get_accounts():
    """获取所有账号列表

    Returns:
        ResponseModel: 账号列表
    """
    try:
        accounts = account_service.get_all_accounts()
        return ResponseModel.success(data=accounts, message="获取成功")
    except Exception as e:
        logger.error(f"获取账号列表失败: {e}")
        return ResponseModel.error(code=500, message=str(e))


@router.get("/{account_id}", response_model=ResponseModel[YunkeAccountResponse])
async def get_account(account_id: int):
    """获取单个账号信息

    Args:
        account_id: 账号ID

    Returns:
        ResponseModel: 账号信息
    """
    result = account_service.get_account_by_id(account_id)
    if not result:
        return ResponseModel.error(code=404, message="账号不存在")

    account, company = result
    return ResponseModel.success(
        data=account_service._account_to_response(account, company),
        message="获取成功",
    )


@router.post("", response_model=ResponseModel[YunkeAccountResponse])
async def create_or_update_account(data: YunkeAccountCreate):
    """创建或更新账号（Upsert）

    如果手机号+公司代码已存在则更新密码，否则创建新账号

    Args:
        data: 账号数据

    Returns:
        ResponseModel: 账号信息
    """
    try:
        account, is_new = account_service.create_or_update_account(data)
        message = "创建成功" if is_new else "更新成功"
        return ResponseModel.success(data=account, message=message)
    except Exception as e:
        logger.error(f"创建/更新账号失败: {e}")
        return ResponseModel.error(code=500, message=str(e))


@router.put("/{account_id}", response_model=ResponseModel[YunkeAccountResponse])
async def update_account(account_id: int, data: YunkeAccountUpdate):
    """更新账号信息

    Args:
        account_id: 账号ID
        data: 更新数据

    Returns:
        ResponseModel: 更新后的账号信息
    """
    try:
        account = account_service.update_account(account_id, data)
        if not account:
            return ResponseModel.error(code=404, message="账号不存在")

        return ResponseModel.success(data=account, message="更新成功")
    except Exception as e:
        logger.error(f"更新账号失败: {e}")
        return ResponseModel.error(code=500, message=str(e))


@router.delete("/{account_id}", response_model=ResponseModel)
async def delete_account(account_id: int):
    """删除账号

    Args:
        account_id: 账号ID

    Returns:
        ResponseModel: 删除结果
    """
    success = account_service.delete_account(account_id)
    if not success:
        return ResponseModel.error(code=404, message="账号不存在")

    return ResponseModel.success(message="删除成功")


@router.post("/{account_id}/login", response_model=ResponseModel)
async def login_account(account_id: int):
    """手动触发账号登录

    Args:
        account_id: 账号ID

    Returns:
        ResponseModel: 登录结果
    """
    result = await account_service.auto_login(account_id)

    if result["success"]:
        return ResponseModel.success(data=result.get("data"), message=result["message"])
    else:
        return ResponseModel.error(code=400, message=result["message"])


@router.get("/{account_id}/status", response_model=ResponseModel)
async def check_status(account_id: int):
    """检查账号状态

    Args:
        account_id: 账号ID

    Returns:
        ResponseModel: 状态信息
    """
    result = await account_service.check_account_status(account_id)

    return ResponseModel.success(
        data={
            "valid": result["valid"],
            "status": result.get("status"),
            "last_login": result.get("last_login"),
        },
        message=result["message"],
    )
